doc/release-versioning.md

```diff
@@ -0,0 +1,151 @@
+# Versioning
+
+The following is a concise explanation of how Operator SDK versions are determined. The Operator SDK versioning follows [semantic versioning][link-semver] standards.
+
+## Milestones
+
+Operator SDK [milestones][link-github-milestones] represent changes to the SDK spanning multiple issues, such as a design change. Milestones help SDK developers determine how close we are to new releases, either major or minor; a milestone must be completed before a new version is released. Milestones and their involved issues are determined by maintainers.
+
+Milestone labels have the form: `milestone-x.y.0`, where `x` and `y` are major and minor SDK versions, respectively. This particular milestone demarcates the SDK `x.y.0` release; once issues within this milestone are addressed, the release process can begin.
+
+## Major versions
+
+Major version changes can break compatibility between the previous major versions; they are not necessarily backwards or forwards compatible. SDK change targets include but are not limited to:
+- `operator-sdk` command and sub-commands
+- Golang API
+- Formats of various yaml manifest files
+
+## Minor versions
+
+Minor version changes will not break compatibility between the previous minor versions; to do so is a bug. SDK changes will involve addition of optional features, non-breaking enhancements, and *minor* bug fixes identified from previous versions.
+
+### Creating a minor version branch
+
+We expect to release patches for minor releases, so we create a patch trunk to branch from. The naming convention follows "v2.1.x", where the major version is 2, minor is 1, and "x" is a patch version placeholder.
+
+```bash
+$ git checkout \
+    -b "v${MAJOR_VERSION}.${NEW_MINOR_VERSION}.x" \
+    tags/"v${MAJOR_VERSION}.${NEW_MINOR_VERSION}.0"
+$ git push git@github.com:operator-framework/operator-sdk.git \
+    "v${MAJOR_VERSION}.${NEW_MINOR_VERSION}.x"
+```
+
+## Patch versions
+
+Patch versions changes are meant only for bug fixes, and will not break compatibility of the current minor version. A patch release will contain a collection of minor bug fixes, or individual major and security bug fixes, depending on severity.
+
+### Creating a patch version branch
+
+As more than one patch may be created per minor release, patch branch names of the form "${MAJOR_VERSION}.${MINOR_VERSION}.${PATCH_VERSION}" will be created after a bug fix has been pushed, and the bug fix branch merged into the patch branch only after testing.
+
+```bash
+$ git checkout "v${MAJOR_VERSION}.${MINOR_VERSION}.x"
+$ git checkout -b "cherry-picked-change"
+$ git cherry-pick "$GIT_COMMIT_HASH"
+$ git push origin "cherry-picked-change"
+```
+
+# Releases
+
+Making an Operator SDK release involves:
+- Tagging and signing a git commit and pushing the tag to GitHub.
+- Building a release binary, signing the binary, and uploading both binary and signature to GitHub.
```

Comment:
As part of this, I would like to see discussion about releasing ansible operator base image and helm operator along with the binary. 

I think that we should start thinking about all of these things as 1 project and not separate entities with their own release timelines or procedures.

In this case, I believe that a tagged version of the ansible-operator base image (and maybe the binary as well?) should be built, and pushed to a shared quay.io repo. I would like this to happen at the exact same time that operator-sdk release is happening. 

I believe the helm-operator should have a published binary and base-image. I don't know enough about how it is expected to be used to comment much further.

Reply:
@spahl @AlexNPavel @robszumski can you comment on this? Not sure if ansible image packaging has been discussed outside of this PR. This suggestion sounds reasonable to me but needs its own PR.

Reply:
We should do thin in a separate PR. I do agree that it should be tracked and released at the same time. Let's create an issue to not forget.

Reply:
#555